### ISSUE TITLE
Dynamic resolver settings

### ISSUE THREAD

**Issue by ekmungai** [xcash/bootstrap-autocomplete#69]
Problem: I have a dynamic form and would like to update the data-url (api endpoint) of the autocomplete based on an earlier input.

Solution: I would like an additional manageApi command that re-initializes the resolver with new user provided settings.

Alternatives: I tried to use Vue to bind the data-url attribute of the search input but whereas the vue instance data object would update the element, the reolver maintaines the same url it was initialized with.

Context: I'm using Vue.js to manage the form.

**Comment by xcash** [comment]
Hi @ekmungai 
The correct way to do what you're asking is to manage your own resolver. 
```
$('.advancedAutoComplete').autoComplete({
    resolver: 'custom',
    events: {
        search: function (qry, callback) {
            // let's do a custom ajax call
            $.ajax(
                'testdata/test-dict.json',
                {
                    data: { 'qry': qry}
                }
            ).done(function (res) {
                callback(res.results)
            });
        }
    }
});
```

Btw, I'll evaluate your PR for the new api cmd. :)

**Comment by ibrahimab** [comment]
Wouldn't `events.search` option logically be moved to `resolver` option? That way you don't need ajax|custom option and people can provide their own mechanism for data retrieval. 

And optionally `resolver` could better be named `repository` or `provider` right?

**Comment by xcash** [comment]
Hi @ibrahimab  and thank you for pointing this out.

> Wouldn't `events.search` option logically be moved to `resolver` option? That way you don't need ajax|custom option and people can provide their own mechanism for data retrieval.

That's because the "resolving" procedure is an event chain and you can customize every step.

`typed -> searchPre -> search -> searchPost`

As I see a lot of question on this topic I think I need to improve the docs! 

> And optionally `resolver` could better be named `repository` or `provider` right?

Yeah. That's the main problem of developers: naming variables :D 
What led to "resolve" was that it was a promise. 
And that it encapsulates a service that resolves a query... :)
Btw I can't change it now with projects relying on that name.. :)

**Comment by xcash** [comment]
Merged #70
